arccnet/data_generation/timeseries/sdo_processing.py

```diff
@@ -795,3 +845,60 @@ def l4_file_pack(aia_paths, hmi_paths, dir_path, rec, out_table, anim_path):
 
     shutil.copy(anim_path, f"{dir_path}/data/{rec}")
     out_table.write(f"{dir_path}/data/{rec}/{rec}.csv", overwrite=True)
+
+
+def pad_map(map, long, targ_width):
+    r"""
+    Pads the map data of a submap which is narrower than the specified submap width due to reaching the edge of the image array.
+
+    Parameters
+    ----------
+        map : `sunpy.map`
+            The sunpy submap to be resized.
+        long : `float` or `int`
+            The longitude of the active region, used only to check which side of the disk it is located.
+        targ_width : `int`
+            The target width of the submaps within the data run.
+
+    Returns
+    ----------
+        new_map : `sunpy.map`
+            The new, padded submap.
+    """
+    x_dim = map.dimensions[0].value
+    diff = int(targ_width - x_dim)
+    if long > 0:
+        data = np.pad(map.data, ((0, 0), (diff, 0)), constant_values=np.nan)
+    else:
+        data = np.pad(map.data, ((0, 0), (0, diff)), constant_values=np.nan)
+    new_map = sunpy.map.Map(data, map.meta)
+    return new_map
+
+
+def flare_check(start, end, ar_num, table):
```

Comment:
As we chatted about the meeting think this may need to be updated to include the flare class and also need some tests